ethcore/src/state/account.rs

```diff
@@ -382,7 +383,12 @@ impl Account {
 	/// Commit the `storage_changes` to the backing DB and update `storage_root`.
 	pub fn commit_storage(&mut self, trie_factory: &TrieFactory, db: &mut HashDB<KeccakHasher>) -> TrieResult<()> {
 		let mut t = trie_factory.from_existing(db, &mut self.storage_root)?;
-		for (k, v) in self.storage_changes.drain() {
+		let old_storage_changes = {
+			let mut tmp = im::HashMap::new();
+			::std::mem::swap(&mut tmp, &mut self.storage_changes);
```

Comment:
This is not quite related to this specific PR, but I just wonder, what will happen if it'll fail to commit? Should we concern about this?

Reply:
Then that error will be propagate out and cause the block import to fail. I think we handled that correctly in `Client`.